ScanbotSDKExamples/MAUI/ClassicComponent.Maui/Platforms/iOS/CustomViews/BarcodeCameraView_iOS.cs

```diff
@@ -0,0 +1,152 @@
+﻿using BarcodeSDK.MAUI.Models;
+using ClassicComponent.Maui.CustomViews;
+using CoreGraphics;
+using DocumentSDK.MAUI.iOS;
+using ScanbotSDK.iOS;
+using UIKit;
+
+namespace ClassicComponent.Maui.Platforms.iOS.CustomViews
+{
+    public class BarcodeCameraView_iOS : UIView
+	{
+        BarcodeCameraViewHandler barcodeCameraViewHandler;
+        SBSDKBarcodeScannerViewController cameraViewController;
+        public BarcodeCameraView_iOS(CGRect frame) : base(frame) { }
+
+        internal async void ConnectHandler(BarcodeCameraViewHandler barcodeCameraViewHandler)
+        {
+            this.barcodeCameraViewHandler = barcodeCameraViewHandler;
+            var visibleViewController = await CheckViewControllerAvailable(barcodeCameraViewHandler);
+            if (visibleViewController != null)
+            {
+                cameraViewController = new SBSDKBarcodeScannerViewController(visibleViewController, barcodeCameraViewHandler.PlatformView);
+                cameraViewController.Delegate = new BarcodeScannerDelegate
+                {
+                    OnDetect = HandleBarcodeScannerResults
+                };
+                cameraViewController.BarcodeImageGenerationType = SBSDKBarcodeImageGenerationType.None;
+                SetConfigurations();
+            }
+        }
+
+        private async Task<UIViewController> CheckViewControllerAvailable(BarcodeCameraViewHandler barcodeCameraViewHandler)
+        {
+            var retryCount = 5;
+
+            UIKit.UIViewController viewController = null;
+            do
+            {
+                await Task.Delay(500);
+                viewController = AppDelegate.ExtractViewController(barcodeCameraViewHandler?.PlatformView?.Window);
+                var message = viewController != null ? "available" : "unavailable";
+                System.Diagnostics.Debug.WriteLine("ViewController is " + message);
+                retryCount++;
+            } while (viewController == null || retryCount < 5);
+            return viewController;
+        }
+
```

Comment:
We definitely need a better way to do this. Retrying at time intervals is a very sketchy workaround, are we 100% confident there is no other proper way to retrieve the ViewController?

If the answer is yes, then I still have some notes:

`CheckViewControllerAvailable` doesn't check if the ViewController is available. It should be called something like `GetTopViewController` or `TryGetTopViewController`

Also, the function doesn't need to live inside the same class, it can be placed under ViewUtils.cs or something similar, as a static method that accepts a UIWindow as an argument, and it can then be called like this:

```csharp
if (barcodeCameraViewHandler?.PlatformView?.Window is UIWindow window) {
    var viewController = await ViewUtils.TryGetTopViewController(window);
}
```


Also the `ExtractViewController` method:

- Serves no purpose in AppDelegate.cs; it should be moved somewhere else, like ViewUtils.cs
- It should be called GetTopViewController() or GetCurrentViewController(); it's not an extraction, and just `ViewController` is too generic

Other notes:

- The retryCount and delay time in ms should have constant values defined at the top-level of the function OR even better, include them in the method signature as arguments with default values: 

```csharp
public static async Task<UIViewController> TryGetTopViewController(UIWindow window, int retryCount = 5, int retryIntervalMs = 500)
```

- There should be a comment regarding the workaround to explain why we are resorting to this atypical approach
- `System.Diagnostics.Debug.WriteLine` logs should be removed

Reply:
Fixed:
- Moved the methods to ViewUtils in iOS.
- Renamed them and removed the logs.

Reply:
@marcosaia1812 
Regarding the approach of accessing of ViewController: 

We used to access the ViewController in a similar way for Xamarin Forms CustomRenderer. 
The same is used here in this approach of MAUI Handlers. 
The situation is:
- The Common View is loaded in the initialisation/constructor of the Page(Maui page)
- So, Till then the Page is not fully initialised and so not available in the navigation stack 
- That is the reason why it's corresponding ViewController is also not available.
So we give it some time to load and get available in the navigation/app hierarchy.